src/spec/express-route-adapter.spec.js

```diff
@@ -0,0 +1,50 @@
+import { jest } from "@jest/globals";
+import { adaptRoute } from "../express-route-adapter.js"
+
+describe("adaptRoute()", () => {
+  describe("when receiving a controller", () => {
+    const num1 = 2;
+    const num2 = 3;
+    const operation = 'sum';
+    const operationRequest = {
+      num1,
+      num2,
+      operation
+    }
+    const mockController = jest.fn(() => {
+      return {
+        operands: [num1, num2],
+        result: 5,
+        operation
+      }
+    });
+    const mockRequest = {
+      query: {
+        num1,
+        num2
+      },
+      params: {
+        operation
+      }
+    }
+    const mockResponse = {
+      send() {
+        return 'Response sended!'
+      }
+    }
+    const responseSpy = jest.spyOn(mockResponse, 'send')
+
+    it("returns a function", () => {
+      expect(adaptRoute(mockController) instanceof Function).toBeTruthy();
+    });
```

Comment:
Esse negócio de quebrar o teste demais que nem no Ruby não é muito comum em js

<img src="https://media1.tenor.com/m/fBvQV_5Lp6UAAAAC/we-dont-do-that-here-black-panther.gif" />

Reply:
O `it` abaixo já o suficiente pra validar que retorna uma função.